openwisp_controller/config/whois/service.py

```diff
@@ -92,6 +105,19 @@ def _need_whois_lookup(self, new_ip):
 
         return self.is_whois_enabled
 
+    def _need_approximate_location_management(self, new_ip):
+        """
+        Used to determine if Approximate locations need to be created/updated
+        or not during WHOIS lookup.
+        """
+        if not self.is_valid_public_ip_address(new_ip):
+            return False
+
+        if not self.is_whois_enabled:
+            return False
```

Comment:
I think based on our discussion we have decided to do cheap checks first. `is_valid_public_ip_address` will require an API call and should be done after checking if whois is enabled or not